api/envoy/service/ratelimit/v3/rls.proto

```diff
@@ -131,4 +132,7 @@ message RateLimitResponse {
 
   // A list of headers to add to the request when forwarded
   repeated config.core.v3.HeaderValue request_headers_to_add = 4;
+
+  // A response body to send to the downstream client when the response code is not OK.
+  string body = 5;
```

Comment:
Should this be a `DataSource` (e.g. see what's done in `DirectResponseAction`)? 

There are other places in the API where we even allow things like log template values to be included and JSON returned,  e.g. `SubstitutionFormatString` in HCM local reply mapper. That could be another reasonable way to future proof and align with the existing precedent for this kind of body.

Reply:
This body comes from the external ratelimit service so I think the better comparison would be the auth response message in the ext_authz filter.

Reply:
To add more context, I think the `DirectResponseAction` is used to configure a response generated locally at Envoy. That means you can configure a data source local to the Envoy container and have access to variables within Envoy's `SubstitutionFormatString`. Since this `body` is configuring a response generated externally in the `RateLimitService` as part of its `RateLimitResponse`, it makes sense to represent that as just a `string` on the wire as protobuf (as we do in  ext_authz https://github.com/envoyproxy/envoy/blob/master/api/envoy/service/auth/v3/external_auth.proto#L59).

Reply:
Right, makes sense!

Reply:
Can we please make this `bytes` and not a string? We have endless problems where we have things that proto can't encode in a string field and technically body can be raw bytes.

Reply:
Sounds good. We'll want to make the same change to ext_authz by adding a new `body_bytes` field and deprecating the existing string `body` field. For consistency, I'll change this field to also be `body_bytes` with type `bytes`. This way ext_authz and ratelimit will have a similar UX (which was part of the goal of this PR). Does that work?

Reply:
Up to you whether you want to do the deprecation dance on extauth but for new API I would definitely do it with bytes.